### DISCUSSION TITLE
OpenResty和面板网络问题

### DISCUSSION THREAD

**Discussion by JAINKRE** [1Panel-dev/1Panel#6997] [Q&A]
### 1Panel 版本

v1.10.20-lts

### 问题描述

问题1，通过面板终端连接不上：

![image](https://github.com/user-attachments/assets/d0ca6e08-32af-412a-824a-a0bcc86399b7)

> 终端连接试了三个地址，全部连不上
> 127.0.0.1 | root | 22 | 默认
> 192.168.3.88 | root | 22 | 默认
> 192.168.3.16 | root | 22 | 默认

192.168.3.88和192.168.3.16均为主机地址，且为万兆网卡。

问题2，OpenResty部署的网站使用默认配置的地址均打不开且OpenResty中没有日志记录：
![image](https://github.com/user-attachments/assets/a79c335d-9bd6-4c5b-a51e-40565b005f76)

访问：
![image](https://github.com/user-attachments/assets/56902c1d-c12e-4f86-9770-2edc80b54f67)
>已经确定，这些网站并没有被任何修改过包括配置，且知道昨天还可以正常使用，防火墙正常放开端口

### 重现步骤

起初，怀疑是设备的网卡以及内部局域网出现了问题，但实际排除后基本排除

连接局域网，另一台设备ping:
![image](https://github.com/user-attachments/assets/03c3f1ac-403f-4bf4-bc86-a2783dfc6e96)

并且，访问的面板地址就是主机局域ip地址，是可以正常访问的。

然后怀疑是不是`192.168.3.88`地址出现了问题，但是设备该ip其他端口访问正常：
![image](https://github.com/user-attachments/assets/593d8fd2-4a03-40ff-9b71-a23743a5d8da)
成功：
![image](https://github.com/user-attachments/assets/1a126972-ec68-4467-85be-4aeddb3a7999)

为了确定问题是OpenResty中出现的（所有OpenResty挂载的网站均不可访问），特意拉去了我们使用nginx作为静态环境的容器，看到是可以正常访问的：
![image](https://github.com/user-attachments/assets/0069b186-da2a-43f0-a69c-550dc2c2d126)
![image](https://github.com/user-attachments/assets/3e53d8bd-8938-4a26-aa1f-89902c257877)

然而，这个问题诡异的地方在于，设备网卡有两个ip，使用另一个`192.168.3.16`却可以正常访问OpenResty中代理的站点：
![image](https://github.com/user-attachments/assets/f6132a01-6bd5-4581-a21b-5f5255a7c31b)
换`192.168.3.88`失败
![image](https://github.com/user-attachments/assets/4af58640-8361-4930-9d04-8dca4ab0ea03)

为了防止是浏览器（使用的chrome）缓存等问题导致，特意换了firefox和firefox隐私浏览，以及清空缓存，均出现`192.168.3.88`不可访问`192.168.3.16`可以访问的问题

更奇怪的是，当使用机器中运行的Windows远程连接的时候，`192.168.3.88`和`192.168.3.16`却同时可访问：
![image](https://github.com/user-attachments/assets/4d198d77-9989-443d-af25-5e497279c99b)

### 期待的正确结果

目前可以确定的是，局域网内出问题的`192.168.3.88`地址只出现OpenResty管理的所有应用不可访问（直到昨天还是正常的），已经尝试过容器重启，但是因为OpenResty中安装的静态网站过多没敢轻易卸载重装

还有一个办法没有尝试就是重启机器，但目前机器在高负载工作不能重启。因此，先提交一下issue，麻烦社区内帮忙查看一下可能的问题原因。

我怀疑可能是OpenResty出现了某些问题，但也找不到OpenResty日志

**Comment by wanghe-fit2cloud** [comment]
问题一参考 [使用自建 Nginx 反向代理 1Panel，导致主机和容器终端均不能正常使用](https://1panel.cn/docs/faq/system_function/)

**Comment by JAINKRE** [comment]
> 问题一参考 [使用自建 Nginx 反向代理 1Panel，导致主机和容器终端均不能正常使用](https://1panel.cn/docs/faq/system_function/)

并没有使用nginx反向代理，包括问题1里的终端连不上，昨天都是正常的。nginx只是作为容器，拉取应用验证是否为192.168.3.88 ip问题。

**Comment by wanghe-fit2cloud** [comment]
目前我这里也不能准确定位到具体问题，可能存在以下几种情况吧：

1. 检查 OpenResty 的 nginx.conf 配置文件中 listen 指令是否限制了绑定的 IP 地址（建议将 listen 设置为 0.0.0.0:80，让 OpenResty 监听所有网卡）。
2. 检查服务器的防火墙规则（如 iptables 或 firewalld）是否对 192.168.3.88 的访问进行了限制。
3. 在多网卡情况下，系统的路由规则可能会影响请求的路由方向，确保两张网卡的流量都被正确地路由到容器所在的网卡。

以上是几个可能的原因，可以逐一排查，看看是否解决了该问题。

**Comment by wanghe-fit2cloud** [comment]
> > 问题一参考 [使用自建 Nginx 反向代理 1Panel，导致主机和容器终端均不能正常使用](https://1panel.cn/docs/faq/system_function/)
> 
> 并没有使用nginx反向代理，包括问题1里的终端连不上，昨天都是正常的。nginx只是作为容器，拉取应用验证是否为192.168.3.88 ip问题。

浏览器上通过 `IP:端口` 访问的吗？
> 参考： https://github.com/1Panel-dev/1Panel/issues/1552

**Comment by JAINKRE** [comment]
> 目前我这里也不能准确定位到具体问题，可能存在以下几种情况吧：
> 
> 1. 检查 OpenResty 的 nginx.conf 配置文件中 listen 指令是否限制了绑定的 IP 地址（建议将 listen 设置为 0.0.0.0:80，让 OpenResty 监听所有网卡）。
> 2. 检查服务器的防火墙规则（如 iptables 或 firewalld）是否对 192.168.3.88 的访问进行了限制。
> 3. 在多网卡情况下，系统的路由规则可能会影响请求的路由方向，确保两张网卡的流量都被正确地路由到容器所在的网卡。
> 
> 以上是几个可能的原因，可以逐一排查，看看是否解决了该问题。

好的，等机器任务执行结束后我先重启一下设备看看。问题应该不在防火墙规则上，如果限制不可能只限制OpenResty使用的端口号。我觉得大概率问题还是出现在了OpenResty上，总之，我先找找看吧。

**Comment by wanghe-fit2cloud** [comment]
好的，我们先将该 issue 转为 discussion，以便进一步讨论。
